interview-questions.md

```diff
@@ -6,63 +6,108 @@ Answer the following questions. First, without external resources. Challenge you
 
   Your answer:
 
+  Functions are reusable pieces of code. Can have lots of different code inside such as arguments. They are like a giant umbrella of information. You can name each function. 
+
   Researched answer:
+  
+  Functions are blocks of code designed to perform a particular task. It is excevited when "something" invokes it. It always returns an output, can be used many times, and should be as generic as possible.
+  
+  To specify the values you need to call the function by typing the name along with the desired values spaced by commas.
+  
+  Example:
+  function myFunction(value1, value2) {
+	return value1 + value2;
+  }
 
+  myFunction(10, 5);
 
 
 2. What is the difference between map and filter?
 
   Your answer:
+  The difference is that .map() looks at each element in an array while .filter() has a true/false code that your array has to go through.
+
+  .map() is an array method, returns to us a completely new array.
+
+  .filter() is also an array method, like map, but it has a built in if/else statement so it looks for true/false values. It returns back what we "filtered" through it. 
 
   Researched answer:
 
+  They both return a new array. map returns a new array of elements where you have applied some function on the element so that it changes the original element (typically). filter returns a new array of the elements of the original array (with no change to the original values). filter will only return elements where the function you specify returns a value of true for each element passed to the function.
+
+  So map returns the same number of elements as the original, but the element value will be transformed in some way and filter will return the same or less number of elements than the original but not change the original elements’ values.
 
 
 3. What is the difference between map/filter and a for loop?
 
   Your answer:
+  A for loop iterates from a defined start and end point with a defined number of times. Map/filter only go through the array once.
+
 
   Researched answer:
 
+  The big difference is that the conditions for iteration are handled for us. In a for loop, we’re guaranteed to get every element of the array in the right order. We don’t have to manage our own index variable or condition for that variable.
 
 
 4. What is the difference between console.log() and return?
 
   Your answer:
 
+  console.log is used for developers, we shouldn't count on it when defining a function. It's helpful when testing out a piece of code. A return would live inside a function and can be used in reference inside the function.
```

Comment:
Sounds about right! I'd probably mention that console.log really only is useful to logging to the console and wouldn't be used in production, while a return can actually return a value.